index/secondary_index_manager.cc

```diff
@@ -341,4 +346,24 @@ bool secondary_index_manager::is_global_index(const schema& s) const {
     });
 }
 
+std::optional<sstring> secondary_index_manager::custom_index_class(const schema& s) const {
+
+    auto idx = _indices.find(index_name_from_table_name(s.cf_name()));
+
+    if (idx == _indices.end() || !(*idx).second.metadata().options().contains(cql3::statements::index_target::custom_index_option_name)) {
+        return std::nullopt;
+    } else {
+        return (*idx).second.metadata().options().at(cql3::statements::index_target::custom_index_option_name);
+    }
+}
+
+// We pass the feature_service as the supported custom classes will depend on the features
+bool secondary_index_manager::is_custom_class_supported(const gms::feature_service& fs, const sstring& class_name) {
+
+    const static std::unordered_set<std::string_view> classes = {
+        "vector_index",
+    };
```

Comment:
It's better to have this as a map, pointing from the name to the implementation.

Otherwise you'll have to duplicate the name when you look up the implementation.